editor/debugger/debug_adapter/debug_adapter_protocol.h

```diff
@@ -139,7 +139,7 @@ class DebugAdapterProtocol : public Object {
 	void notify_stopped_breakpoint(const int &p_id);
 	void notify_stopped_step();
 	void notify_continued();
-	void notify_output(const String &p_message);
+	void notify_output(const String &p_message, const RemoteDebugger::MessageType& p_type);
```

Comment:
```suggestion
	void notify_output(const String &p_message, RemoteDebugger::MessageType p_type);
```
Simple type, pass by value

Reply:
Isn't it better to pass by reference instead? Even it it's a very small datatype, doesn't it save a copy?

Reply:
Nope, it adds overhead and prevents passing in a register simply, remember it could be changed elsewhere if passed by reference and the compiler can't optimize easily if you do, it can't assume it's unchanged

Reply:
Yeah, that makes sense. Thank you for the detailed explanation!

Reply:
Fixed :+1: